### ISSUE TITLE
Game crashed with message: Attempt to visit object that is being visited by another hero! Disaster happened. Reason: Can not visit object that is being visited

### ISSUE THREAD

**Issue by nerozero** [vcmi/vcmi#4848]
**Describe the bug**
After ending my move - game crashed with message: 
Attempt to visit object that is being visited by another hero!
Disaster happened.
Reason: Can not visit object that is being visited

**Game logs**
Please attach game logs: [VCMI_Client_log.txt](https://github.com/user-attachments/files/17571503/VCMI_Client_log.txt)

**To Reproduce**
 1. No specific recipe, rarely happening on the AI move, map is still unexplored back

**Expected behavior**
Not to crash, i guess...

**Actual behavior**
Crashing rarely, 2nd time happened. Compiled from git today.

**Did it work earlier?**
yes.

**Screenshots**
No screenshots... Game crashed... 

**Version**
 - OS: Linux
 - Version: VCMI 1.6.0.3a45e6dae13317afeb4b4e4fc8427dc79d267335

**Additional context**
Please fix the log file names: `VCMI_client.txt` -> `VCMI_Client_log.txt` .

**Comment by IvanSavenko** [comment]
Please also upload save before crash and map.
Seems to be related to AI actions & map events

**Comment by Povelitelh3** [comment]
Well, the mistake of the FIXME 
> Attempt to visit object that is being visited by another hero!
 turned into a crash :)

**Comment by nerozero** [comment]
@IvanSavenko Here is the Save, Crash log and the map itself in the archive.
[0_shibby_map!!!_20241030T111608.zip](https://github.com/user-attachments/files/17572857/0_shibby_map._20241030T111608.zip)

**Comment by GrafDeVan** [comment]
> [@IvanSavenko](https://github.com/IvanSavenko) Here is the Save, Crash log and the map itself in the archive. [0_shibby_map!!!_20241030T111608.zip](https://github.com/user-attachments/files/17572857/0_shibby_map._20241030T111608.zip)

AI was redesigned to NK2. Many bugs were fixed. 
Please try the latest version 1.7.4 https://vcmi.eu/download/ or latest develop https://download.vcmi.eu/branch/develop/

**Comment by GrafDeVan** [comment]
There's definitely a bug in this map. You start in a dungeon, and on the second day, the battle academy is built via an event, and the AI ​​prompts the player to choose which skill to take. The game freezes.

<img width="1219" height="789" alt="Image" src="https://github.com/user-attachments/assets/4e0f14f8-5c47-4d38-99ae-d7a28f129c13" />

<img width="825" height="439" alt="Image" src="https://github.com/user-attachments/assets/9ae2bbd2-6519-41e0-b74b-1af7f2b78f0c" />

save [experience.zip](https://github.com/user-attachments/files/28680964/experience.zip)

Win 10 x64, dev 0419a02567ec986c6074e39fa45eeb4fac1946a6

**Comment by GrafDeVan** [comment]
Second case is 

```
[2026-Jun-07 15:32:11.870374] DEBUG [runServer] global - Granting reward 0. Message says: Посетив Боевую академию, вы узнали много нового (+1000 опыта).
[2026-Jun-07 15:32:11.871374] DEBUG [runServer] global - Hero hero.core.sephinroth.name visits Дипшэдоу, Подземелье (98)
[2026-Jun-07 15:32:11.871374] ERROR [runServer] global - Attempt to visit object that is being visited by another hero!
```

And game close 
save [close.zip](https://github.com/user-attachments/files/28681071/close.zip)

[VCMI_Client_log.txt](https://github.com/user-attachments/files/28681096/VCMI_Client_log.txt)

**Comment by GrafDeVan** [comment]
@GeorgeK1ng your PR 7437 doesn't help in this situation ^ Or it is other case?

**Comment by GeorgeK1ng** [comment]
> [@GeorgeK1ng](https://github.com/GeorgeK1ng) your PR 7437 doesn't help in this situation ^ Or it is other case?

Yes, thats different case, PR 7437, fixes regression, created by fixing flickering hero level-up window from my previous PR.

**Comment by MichalZr6** [comment]
Both cases from `experience.zip` and `close.zip` are fixed in #7308

**Comment by GrafDeVan** [comment]
@GeorgeK1ng
